src/main/java/com/limechain/trie/structure/nibble/Nibbles.java

```diff
@@ -54,6 +54,10 @@ public static Nibbles of(Iterator<Nibble> nibbles) {
         return new Nibbles(IteratorUtils.toList(nibbles));
     }
 
+    public static Nibbles of(byte[] bytes) {
```

Comment:
I'd change this method name to `fromBytes`, since those `of` factory methods have the semantics of 'construct me from this representation of nibbles', but here an additional iteration logic (splitting bytes in half) occurs.